### ISSUE TITLE
sometimes nothing happen when you click a menu item

### ISSUE THREAD

**Issue by Jimmy-Z** [Jimmy-Z/Linkle#1]
this is actually my note on resolving this issue.

TL;DR
===
so based on this sample, contextMenus.create only needs to be (and apparently should only be) called onInstalled
https://developer.chrome.com/extensions/examples/api/contextMenus/event_page/sample.js
and of course when the config got updated

and onClicked.addListener should be called everytime it loads
https://developer.chrome.com/extensions/event_pages#best-practices

History
===
the old extension was working OK with background page and hard coded config, then I migrated to event page and added option page, things got complicated and this is my first Chrome extension.
different ways I've tried:
1) do everything when onInstalled fires. the mistake I made is I should have called onClicked.addListener on reload, I think this is absurd, if chrome can remember all contextMenu.create why not this event listener? anyway I should have figured this out earlier since there is no onClicked.removeListener, how stupid I was. and another stupid mistake was since I didn't see any runtime.onReload, I'm guessing by "install" it means  every time when chrome starts it "install" the extension, that was incorrect, actually everything on the top level of the script runs on reload so there is no runtime.onReload.  
2) do everything on reload, works in test, later I noticed onClick was not fired after event page unloaded, the event page loaded when menu got clicked, but onClick was not called, a situation not triggered during earlier testing, I guess I tried everything quickly, event page doesn't got unloaded. and this is major PITA to debug, when I opened background page console and try to reproduce the bug, it just won't happen.
and I still don't understand the logic behind this, It's not perfect but why it doesn't work? so the user clicked the context menu, chrome load the event page for the event handler, I wrongfully recreated all contextMenu, then chrome decided not to call the event handler? it was my fault not to read the entire fucking manual, but still, this is anti intuition, at least needs to be documented.
3) current, contextMenu.create onInstalled, and onClicked.addListener on reload, I really hope this is the final time I got it right.
